ga/client/bitrateadaptation.cpp

```diff
@@ -162,120 +129,126 @@ static unsigned int bbr_rtt_head = 0;
  *		Leaves once Excess created by startup is drained
  * 2 : Probe / steady state
  */
-static int bbr_state = -1; 
-
-// 3 round window for detecting plateaus in startup
-static unsigned int bbr_startup_prev1 = 0;
-static unsigned int bbr_startup_prev2 = 0;
-
-#define BBR_PROBE_INTERVAL_US (4 * 1000 * 1000)
-
-static struct timeval bbr_prev_probe;
-
-#define BBR_BITRATE_MINIMUM 50
-#define BBR_BITRATE_MAXIMUM 30000
-
-static int bbr_bitrate = 200;
-
-/*
-void
-handle_bbrreport(ctrlmsg_system_t *msg) {
-	// Parse network properties
-	ctrlmsg_system_bbrreport_t *msgn = (ctrlmsg_system_bbrreport_t*) msg;
-	unsigned int latest_rtt = 0;
-	m_server->ioctl(GA_IOCTL_CUSTOM, sizeof(unsigned int *), &latest_rtt);
-	
-	unsigned int rtProp = UINT_MAX;
-	if (latest_rtt == 0) {
-		return;
-	} else {
-		bbr_rtt[bbr_rtt_head].rtt = latest_rtt;
-		bbr_rtt_head = (bbr_rtt_head + 1) % BBR_RTT_MAX;
-
-		int seek = (bbr_rtt_head + BBR_RTT_MAX - BBR_RTT_WINDOW_SIZE) % BBR_RTT_MAX;
-		while (seek != bbr_rtt_head) {
-			if (bbr_rtt[bbr_rtt_head].rtt < rtProp) {
-				rtProp = bbr_rtt[bbr_rtt_head].rtt;
-			}
-			seek = (seek + 1) % BBR_RTT_MAX;
-		}
-
-		// ga_error("RTProp: %u ms RTT: %u ms rcvrate: %d\n", rtProp * 1000 / 65536, latest_rtt * 1000 / 65536, msgn->rcvrate);
-	}
-	// Determine gain based on state
+#define BBR_PROBE_INTERVAL_US (4 * 1000 * 1000) // 4 seconds
+
+typedef struct bbr_state_s {
+	int stage;
+	unsigned int start_0;
+	unsigned int start_1;
+	struct timeval prev_probe;
+	int rtprop; // Time delta values in microseconds
+	int latest_rtt;
+	int bitrate;
+} bbr_state_t;
```

Comment:
lines 132-142: this stuff should be in a header file. Also, all of the fields in struct bbr_state_s should be commented